src/lib/KeyStore.js

```diff
@@ -62,16 +74,21 @@ class KeyStore {
             return null;
         }
 
-        if (!keyRecord.encrypted) {
-            return new Key(keyRecord.secret, keyRecord.type);
+        if (!KeyStore.isEncrypted(keyRecord)) {
+            const purposeId = new Nimiq.SerialBuffer(keyRecord.secret).readUint32();
+            const secret = purposeId === Nimiq.Entropy.PURPOSE_ID
```

Comment:
I am not sure about it, but it feels to me like there should be something like
```suggestion
                : (purposeId === Nimiq.PrivateKey.PURPOSE_ID 
                    ? new Nimiq.PrivateKey(keyRecord.secret.subarray(4))
                    : null; // I don't know
```

Reply:
Nope, because private keys are always encrypted.

Reply:
I made it now use `keyRecord.type` instead of the purposeID, and also check for PrivateKey and create an Entropy in all other cases (as suggested here).